homeassistant/components/tuya/alarm_control_panel.py

```diff
@@ -0,0 +1,149 @@
+
+from . import HomeAssistantTuyaData
+from .base import TuyaEntity
+from .const import DOMAIN, TUYA_DISCOVERY_NEW, DPCode
+
+_LOGGER = logging.getLogger(__name__)
+
+ICON = "mdi:security"
+
+SUPPORT_ALARM_ARM_HOME = 1
+SUPPORT_ALARM_ARM_AWAY = 2
+SUPPORT_ALARM_ARM_NIGHT = 4
+SUPPORT_ALARM_TRIGGER = 8
+SUPPORT_ALARM_ARM_CUSTOM_BYPASS = 16
+SUPPORT_ALARM_ARM_VACATION = 32
+
+
+@dataclass
+class TuyaAlarmEntityDescription(AlarmControlPanelEntityDescription):
+    """Describe an Tuya cover entity."""
+
+    battery_percentage: DPCode | None = None
+
+
+# All descriptions can be found here:
+# https://developer.tuya.com/en/docs/iot/standarddescription?id=K9i5ql6waswzq
+ALARM: dict[str, tuple[TuyaAlarmEntityDescription, ...]] = {
+    # Alarm
+    # https://developer.tuya.com/en/docs/iot/categorymal?id=Kaiuz33clqxaf
+    "mal": (
+        TuyaAlarmEntityDescription(
+            key="master_mode",
+            name="Alarm",
+            battery_percentage=DPCode.BATTERY_PERCENTAGE,
+        ),
+    ),
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up Tuya alarm dynamically through Tuya discovery."""
+    hass_data: HomeAssistantTuyaData = hass.data[DOMAIN][entry.entry_id]
+    _LOGGER.info(hass_data)
+
+    @callback
+    def async_discover_device(device_ids: list[str]) -> None:
+        """Discover and add a discovered Tuya siren."""
+        entities: list[TuyaAlarmEntity] = []
+        for device_id in device_ids:
+            device = hass_data.device_manager.device_map[device_id]
+            if descriptions := ALARM.get(device.category):
+                for description in descriptions:
+                    if (
+                        description.key in device.function
+                        or description.key in device.status
+                    ):
+                        entities.append(
+                            TuyaAlarmEntity(
+                                device, hass_data.device_manager, description
+                            )
+                        )
+        async_add_entities(entities)
+
+    async_discover_device([*hass_data.device_manager.device_map])
+
+    entry.async_on_unload(
+        async_dispatcher_connect(hass, TUYA_DISCOVERY_NEW, async_discover_device)
+    )
+
+
+class TuyaAlarmEntity(TuyaEntity, AlarmControlPanelEntity):
+    """Tuya Alarm Entity."""
+
+    def __init__(
+        self,
+        device: TuyaDevice,
+        device_manager: TuyaDeviceManager,
+        description: TuyaAlarmEntityDescription,
+    ) -> None:
+        """Init Tuya Alarm."""
+        super().__init__(device, device_manager)
+        self.entity_description = description
+        self._attr_unique_id = f"{super().unique_id}{description.key}"
+
+    @property
+    def state(self):
+        """Return the current Alarm State."""
+        return self.device.status.get(DPCode.ALARM_STATE)
```

Comment:
The states of Tuya provides, do not match the states Home Assistant expects.
See: <https://developers.home-assistant.io/docs/core/entity/alarm-control-panel#states>